File: algorithms/java/第N个泰波那契数/第N个泰波那契数.java
// Source : https://leetcode.cn/problems/n-th-tribonacci-number/
// Author : Yanan Wang
// Date   : 2023-06-11

class Solution {
    public int tribonacci(int n) {
        if (0 == n) return 0;
        if (1 == n || 2 == n) return 1;
        int a = 0, b = 1, c = 1, d = 0;
        for (int i = 3; i <= n; i++) {
            d = a + b + c;
            a = b;
            b = c;
            c = d;
        }
        return d;
    }
}

File: algorithms/python/第 N 个泰波那契数/第 N 个泰波那契数.py
# Source : https://leetcode.cn/problems/n-th-tribonacci-number/
# Author : Yanan Wang
# Date   : 2023-10-29

class Solution:
    def tribonacci(self, n: int) -> int:
        if n == 0:
            return 0
        if n <= 2:
            return 1
        a = 0
        b = c = 1
        for i in range(3, n + 1):
            ans = a + b + c
            a, b, c = b, c, ans
        return ans